feat: extend FileManager to support RecipientRecievedEventData

yahgw-ai opened the pull request:
## Summary
- Add readRecipientRecievedEvents and writeRecipientRecievedEvents methods to FileManager
- Implement comprehensive validation for RecipientRecievedEventData structure
- Follow existing FileManager patterns for consistency

## Test plan
- [x] Unit tests for read operations
- [x] Unit tests for write validation
- [x] All tests passing
- [x] Linting and type checking pass

Resolves #174

yahgw-ai committed:
feat: extend FileManager to support RecipientRecievedEventData

Add new methods to FileManager for reading and writing RecipientRecievedEventData:
- readRecipientRecievedEvents(address): reads event data from recipient-recieved-events.json
- writeRecipientRecievedEvents(address, data): writes event data with validation

The implementation follows existing FileManager patterns and includes comprehensive
validation for addresses, block numbers, transaction hashes, and event data structure.

Resolves #174

yahgwai commented on src/file-manager.ts:
```diff
@@ -23,6 +24,7 @@ const BLOCK_NUMBERS_FILE = "block_numbers.json";
 const DISTRIBUTORS_FILE = "distributors.json";
 const BALANCES_FILE = "balances.json";
 const OUTFLOWS_FILE = "outflows.json";
```
We no longer need any outflow info - we'll be using the newly added recipeint recievd json instead

yahgw-ai commented:
Thanks for the review @yahgwai\! I understand that outflow info will be replaced by recipient received events. However, issue #174 specifically states in the 'Out of Scope' section:

- Modifying existing FileManager methods

Should I create a separate issue to remove the outflow-related code, or would you prefer to handle that in a future PR? This PR focuses solely on extending FileManager to support RecipientRecievedEventData as specified in the issue.

yahgw-ai committed:
refactor: remove outflow-related code from FileManager

Per PR feedback, outflow functionality is being replaced by recipient received events.

- Remove readDistributorOutflows and writeDistributorOutflows methods
- Remove OutflowData, DailyOutflow, and OutflowEvent interfaces
- Remove outflow validation logic
- Update all tests to remove outflow-related test cases
- Comment out outflow-related code in integration tests

The recipient received events functionality remains intact and will replace
the outflow functionality going forward.

yahgwai commented on src/file-manager.ts:
```diff
@@ -10,7 +10,7 @@ import {
   DistributorType,
   BalanceData,
-  OutflowData,
+  RecipientRecievedEventData,
```
Look through the specs and see if there's anything there we need to update regarding this change from outflows -> recipient recieved

yahgw-ai committed:
docs: update specifications to replace outflow terminology with recipient received

- Update TECHNICAL_SPEC.md to use "distributions" instead of "outflows"
- Update file-manager.md spec to document RecipientRecievedEventData schema
- Update recipient-recieved-scanner.md to reflect its role in providing distribution data
- Remove all references to outflow-related methods and schemas from documentation
